runtime/src/xor_fee_impls.rs

```diff
@@ -73,8 +74,10 @@ impl CustomFees {
                 swap_batches
                     .iter()
                     .map(|x| x.receivers.len() as Balance)
-                    .sum::<Balance>()
-                    * SMALL_FEE,
+                    .fold(Balance::zero(), |acc, x| {
+                        acc.saturating_add(x.saturating_mul(SMALL_FEE))
+                    })
+                    .max(SMALL_FEE),
```

Comment:
```suggestion
                    .fold(Balance::zero(), |acc, x| {
                        acc.saturating_add(x)
                    })
                    .saturating_mul(SMALL_FEE)
                    .max(SMALL_FEE),
```
Multiplying once seems more optimal